### PR TITLE
prov/efa: Edit the protocol doc

### PR DESCRIPTION
**Pull request by wckzhang**
Contains formatting fixes, table label fixes, grammatical/spelling
improvements, clarified wording, etc.

Signed-off-by: William Zhang <wilzhang@amazon.com>

### CODE REVIEW

**Comment by wckzhang** [comment]
First pass through, there are probably things I missed but I tried to address the issues I saw

File: prov/efa/docs/efa_rdm_protocol_v4.md [hunk 1]
```diff
@@ -2,105 +2,108 @@
...
-This document describes version 4 of EFA RDM communication protocol (protocol v4),
+This document describes version 4 of the Elastic Fabric Adapter (EFA) Reliable Datagram (RDM) communication protocol (protocol v4),
 which is adopted by libfabric EFA provider's RDM endpoint since libfabric 1.10.0 release.
 
 The purpose of this document is to provide a definition of the protocol that is
 not tied to a specific implementation. It is useful to distinguish protocol and
-implementation, because protocol change can cause backward compatibility issue,
-therefore needs to be handled with extra care.
+implementation since protocol change can cause backward compatibility issues
+and needs be be handled with extra care.
```
Comment (wzamazon) [inline prov/efa/docs/efa_rdm_protocol_v4.md:11]: "be be" is a typo?

File: prov/efa/docs/efa_rdm_protocol_v4.md [hunk 2]
```diff
@@ -2,105 +2,108 @@
...
 
 It is organized as the following:
 
-Chapter 1 "Basics" introduces some basic facts/concepts of EFA RDM protocol, including:
+Chapter 1 "Basics" introduces some basic facts/concepts of the  EFA RDM protocol, including:
```
Comment (wzamazon) [inline prov/efa/docs/efa_rdm_protocol_v4.md:15]: extra space?

File: prov/efa/docs/efa_rdm_protocol_v4.md [hunk 3]
```diff
@@ -122,31 +125,31 @@ Table: 1.1 a list of sub-protocols
...
 
-A REQ packet was the 1st packet sender/requester send to the receiver/responder
-in the workflow of a sub-protocol. Each sub-protocol is unique, thus each
-sub-protocol defines its own REQ packet type.
+A REQ packet is the 1st packet the sender/requester sends to the receiver/responder
+in the workflow of a subprotocol. Each subprotocol is unique, thus each
+subprotocol defines its own REQ packet type.
 
-A non-REQ packet is used to by some sub-protocols to transfer additional
+A non-REQ packet is used to by some subprotocols to transfer additional
```
Comment (wzamazon) [inline prov/efa/docs/efa_rdm_protocol_v4.md:143]: I think "used to" should be "used"

File: prov/efa/docs/efa_rdm_protocol_v4.md (hunk 1 above)
Comment (wckzhang) [reply prov/efa/docs/efa_rdm_protocol_v4.md:11]: yeah that should be "to be" not "be be"

File: prov/efa/docs/efa_rdm_protocol_v4.md (hunk 2 above)
Comment (wckzhang) [reply prov/efa/docs/efa_rdm_protocol_v4.md:15]: Will remove

File: prov/efa/docs/efa_rdm_protocol_v4.md (hunk 3 above)
Comment (wckzhang) [reply prov/efa/docs/efa_rdm_protocol_v4.md:143]: Will change

**Review by wzamazon — APPROVED** [review]

**Pull request merged**